archunit/src/main/java/com/tngtech/archunit/library/freeze/ViolationStoreFactory.java

```diff
@@ -0,0 +1,225 @@
+        } catch (Exception e) {
+            String message = String.format("Could not instantiate %s of configured type '%s=%s'",
+                    ViolationStore.class.getSimpleName(), FREEZE_STORE_PROPERTY, violationStoreClassName);
+            throw new StoreInitializationFailedException(message, e);
+        }
+    }
+
+    @VisibleForTesting
+    static class TextFileBasedViolationStore implements ViolationStore {
+        private static final Logger log = LoggerFactory.getLogger(TextFileBasedViolationStore.class);
+
+        private static final Pattern UNESCAPED_LINE_BREAK_PATTERN = Pattern.compile("(?<!\\\\)\n");
+        private File storeFolder;
+        private FileSyncedProperties storedRules;
+
+        @Override
+        public void initialize(Properties properties) {
+            String path = properties.getProperty("default.path", "archunit_store");
+            storeFolder = new File(path);
+            ensureExistence(storeFolder);
+            File storedRulesFile = new File(storeFolder, "stored.rules");
+            log.info("Initializing {} at {}", TextFileBasedViolationStore.class.getSimpleName(), storedRulesFile.getAbsolutePath());
+            storedRules = new FileSyncedProperties(storedRulesFile);
+            checkInitialization(storedRules.initializationSuccessful(), "Cannot create rule store at %s", storedRulesFile.getAbsolutePath());
+        }
+
+        private void ensureExistence(File folder) {
+            checkState(folder.exists() && folder.isDirectory() || folder.mkdirs(), "Cannot create folder %s", folder.getAbsolutePath());
+        }
+
+        private void checkInitialization(boolean initializationSuccessful, String message, Object... args) {
+            if (!initializationSuccessful) {
+                throw new StoreInitializationFailedException(String.format(message, args));
+            }
+        }
+
+        @Override
+        public boolean contains(ArchRule rule) {
+            return storedRules.containsKey(rule.getDescription());
+        }
+
+        @Override
+        public void save(ArchRule rule, List<String> violations) {
+            log.debug("Storing evaluated rule '{}' with violations: {}", rule.getDescription(), violations);
+
+            UUID ruleId = ensureRuleId(rule);
+            File ruleDetails = new File(storeFolder, ruleId.toString());
+            write(violations, ruleDetails);
+        }
+
+        private void write(List<String> violations, File ruleDetails) {
+            String updatedViolations = Joiner.on("\n").join(escaped(violations));
+            try {
+                Files.write(updatedViolations, ruleDetails, UTF_8);
+            } catch (IOException e) {
+                throw new StoreUpdateFailedException(e);
+            }
+        }
+
+        private List<String> escaped(List<String> violations) {
+            return replaceCharacter(violations, "\n", "\\\n");
+        }
+
+        // FIXME: Correct word for 'deescaped'?
```

Comment:
Maybe `unescape`?

Reply:
Yes, but the `escaped(violations)` method should then be `escape(violations)`. Then it's consistent and makes sense, I think.